packages/opentelemetry-instrumentation/src/autoLoaderUtils.ts

```diff
@@ -29,20 +30,28 @@ export function parseInstrumentationOptions(
 ): AutoLoaderResult {
   let instrumentations: Instrumentation[] = [];
   for (let i = 0, j = options.length; i < j; i++) {
-    const option = options[i] as any;
+    const option = options[i];
     if (Array.isArray(option)) {
       const results = parseInstrumentationOptions(option);
       instrumentations = instrumentations.concat(results.instrumentations);
-    } else if (typeof option === 'function') {
+    } else if (isInstrumentationClass(option)) {
```

Comment:
nit: this and line 39 seem like overkill, especially as ```isInstrumentationClass``` is just checking if option is a function and not specifically that is a class / constructor. Especially this this really just causes more code to be generated than necessary (coming from a code / browser support minification perspective)

Reply:
Oh, and don't appear to be related to the listed linting issues.

Reply:
Replacing this function with `typeof option === 'function'` results in the following compile error:
```
Cannot create an instance of an abstract class.ts(2511)
```

The nice thing about this type guard function is that it implicitly casts the `option` to the desired type as well.

The `isInstrumentation` function can be removed though. I just added it to improve the readability of the code (by my personal standards!).

Reply:
That would be a side effect that the function is declaring that the object is of type ```option is new () => T``` (which could be added inline) as this gets compiled away.

So simplistically, you have a function that is kludging this. If you want a function I would suggest that we create this as a helper and call it something like ```isConstructor``` although that would still just check that it's a function.

Reply:
> ... declaring that the object is of type option is new () => T (**which could be added inline**) ...

That's actually a good idea. Can you tell me how to do it? I cannot figure out the syntax to do it in the `if` condition.

Reply:
`if (typeof option === 'function')` is your only option for that I think and just adding an ignore to the lint warning. Personally, I would rather just get rid of this behavior entirely and make the users call `new` themselves. I think this "pass whatever you want and we figure it out" is more confusing than helpful.

Reply:
That one doesn't work. Previously, there was no compile error for that since `option` got casted to `any` at line 32. But, the current implementation needs a cast or a type guard. Other fun fact is that we should tell the compiler that this variable is the constructor for one of the subclasses of `InstrumentationBase` and not `InstrumentationBase` since it is abstract and cannot be instantiated.

Reply:
A function providing the type guard is the only way to do what you're asking. It can't be inlined unless I'm not understanding what @MSNev is suggesting

Reply:
I still think we should only allow `Instrumentation[]` but that would need to be handled in a separate PR since it is breaking and quite a substantial change.

Reply:
It would require a change, but it could be done. The meta package could easily return a list of already new'd instrumentations. The constructor itself doesn't do any patching or have side effects.

Reply:
it's not pretty but something like this should work, hence my suggestion of a generic ```isConstructor``` helper
```
} else if (typeof option === 'function') {
  instrumentations.push(new (option as unknown as (new () => InstrumentationOption))());
}
```

Reply:
I prefer disabling the eslint warning instead of cast to unknown then to something else which only pretends to be typesafe. If the call to `new` fails, then the user will get an error during process start. We can't guard against every possible case.

Reply:
What is the resolution here?

Reply:
It seems like the current solution is not preferrable. I will change it back to use `any` and disable the eslint for that line.